core/api/service/state/requests/get_keys_paged.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/service/state/requests/get_keys_paged.hpp"
+
+#include "scale/scale.hpp"
+
+namespace kagome::api::state::request {
+
+  outcome::result<void> GetKeysPaged::init(
+      const jsonrpc::Request::Parameters &params) {
+    if (params.size() > 4 or params.size() == 1 or params.empty()) {
+      throw jsonrpc::InvalidParametersFault("Incorrect number of params");
+    }
+    auto &param0 = params[0];
+
+    if (not param0.IsString() and not param0.IsNil()) {
+      throw jsonrpc::InvalidParametersFault(
+          "Parameter '[prefix]' must be a hex string");
+    }
+
+    if (param0.IsNil()) {
+      prefix_ = common::Buffer();
+    } else {
+      // here param0 is string
+      auto &&prefix_str = param0.AsString();
+      OUTCOME_TRY(key, common::unhexWith0x(prefix_str));
+      prefix_ = common::Buffer(std::move(key));
+    }
+
+    if (not params[1].IsInteger32()) {
+      throw jsonrpc::InvalidParametersFault(
+          "Parameter '[key_amount]' must be a uint32_t");
+    }
+
+    keys_amount_ = params[1].AsInteger32();
+    if (params.size() == 2) {
+      return outcome::success();
+    }
+
+    // process prev_key param
+    if (not params[2].IsString()) {
+      throw jsonrpc::InvalidParametersFault(
+          "Parameter '[prev_key]' must be a hex string of encoded optional of "
+          "bytes");
+    }
+    OUTCOME_TRY(prev_key, common::unhexWith0x(params[2].AsString()));
+    prev_key_ = common::Buffer{prev_key};
+
+    if (params.size() == 3) {
+      return outcome::success();
+    }
+
+    // process at param
+    if (not params[3].IsString()) {
+      throw jsonrpc::InvalidParametersFault(
+          "Parameter '[at]' must be a hex string of encoded optional of bytes");
```

Comment:
A bit hard to understand what it should be. Maybe something like 'a hex string representation of an encoded optional byte sequence' would be a bit clearer.